### PR TITLE
[FIX] remove strict power-of-two constraint for total_dp

### PR DESCRIPTION
**Pull request by sierraisland**
# Description

Currently, `total_dp` is required to be a power of two due to a strict assertion in the token padding logic. This PR relaxes this constraint by ensuring that the internal `min_token_size` is aligned to the next power of two, regardless of whether the user-provided `dp_size` is a power of two. This maintains the hardware's requirements for memory alignment while supporting arbitrary DP configurations.

**Changes:**
- Used `next_power_of_2` utility in `tpu_inference/runner/utils.py`.
- Updated `TPUModelRunner` initialization to calculate `min_token_size` using this new alignment logic.
- Added comprehensive unit tests to ensure different `dp_size` and `kv_packing` combinations align correctly.

FIX: b/483122466

# Tests

**Unit Tests:** Added `test_min_token_size_alignment` in `tests/runner/test_utils.py`. All tests passed.
   - Command: `pytest tpu-inference/tests/runner/test_utils.py`

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/runner/utils.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ class InferencePhase(Enum):
     AMBIGUOUS = 3
 
 
+def next_power_of_two(n: int) -> int:
```
Comment (kyuyeunk) [inline tpu_inference/runner/utils.py]: nit: palls provides the same exact utility function: https://github.com/jax-ml/jax/blob/fff793a03eabfab02baf8d68f63983cb256f5914/jax/_src/pallas/utils.py#L70

Comment (kyuyeunk) [reply tpu_inference/runner/utils.py]: And also vLLM upstream: https://github.com/vllm-project/vllm/blob/main/vllm/utils/math_utils.py#L15

File: tpu_inference/runner/tpu_runner.py [hunk 2]
```diff
@@ -323,11 +324,18 @@ def _create_single_slice_mesh(self) -> jax.Array:
...
+                mesh_shape,
+                self.devices,
+                allow_split_physical_axes=True,
+            )
+        except (AssertionError, ValueError, RuntimeError) as e:
+            logger.warning(
+                "Physical mesh creation failed (shape=%s, devices=%d). "
+                "Falling back to logical reshape. Error: %s", mesh_shape,
+                len(self.devices), e)
+            return np.array(self.devices).reshape(mesh_shape)
```
Comment (wenxindongwork) [inline tpu_inference/runner/tpu_runner.py:341]: @sierraisland should a mesh be returned here?

File: tpu_inference/runner/tpu_runner.py
Comment (wenxindongwork) [inline tpu_inference/runner/tpu_runner.py]: could you also add the fallback here?

File: tpu_inference/runner/tpu_runner.py (hunk 2 above)
Comment (wenxindongwork) [reply tpu_inference/runner/tpu_runner.py:341]: and curious if you know why `mesh_lib.Mesh` works but `jax.make_mesh` doesn't?

Comment (sierraisland) [reply tpu_inference/runner/tpu_runner.py:341]: Actually, I'd prefer to keep the return type as `np.ndarray` here to remain strictly consistent with JAX's native `mesh_utils.create_device_mesh` and `create_hybrid_device_mesh` APIs.

Comment (sierraisland) [reply tpu_inference/runner/tpu_runner.py:341]: `jax.make_mesh` performs a strict physical topology check to ensure a perfect rectangular grid on the hardware. On an 8-core host, DP=6 is physically 'irregular,' so it fails that check. `mesh_lib.Mesh` is a low-level API that only cares about logical layout, allowing us to bypass the hardware constraints.

Comment (wenxindongwork) [reply tpu_inference/runner/tpu_runner.py:341]: awesome, thanks for the explanation, please free feel to add this in the comments
